pdr_backend/aimodel/aimodel_plotter.py

```diff
@@ -163,113 +178,110 @@ def _plot_aimodel_contour(
 
-    ax.legend(loc=legend_loc)
-    plt.show()
+    return fig
 
 
 @enforce_types
-def plot_aimodel_varimps(
-    varnames: List[str],
-    imps_tup: tuple,
-    fig_ax=None,
-):
+def plot_aimodel_varimps(d: AimodelPlotdata):
     """
     @description
       Bar plot showing rel importance of vars
       Including 95% confidence intervals (2.0 stddevs)
 
     @arguments
-      varnames -- variable names
-      imps_tup -- tuple of (imps_avg, imps_stddev)
-      fig_ax -- None or (fig, ax) to easily embed into existing plot
+      d -- AimodelPlotdata
     """
-    n = len(varnames)
-    imps_avg, imps_stddev = imps_tup
-
-    # re-order in descending imps_avg
-    I = np.argsort(imps_avg)[::-1]
-    imps_avg = imps_avg[I]
-    imps_stddev = imps_stddev[I]
-    varnames = [varnames[i] for i in I]
-
-    # if >40 vars, truncate to top 40+1
-    if n > 40:
-        rest_avg = sum(imps_avg[40:])
-        rest_stddev = np.average(imps_stddev[40:])
-        imps_avg = np.append(imps_avg[:40], rest_avg)
-        imps_stddev = np.append(imps_stddev[:40], rest_stddev)
-        varnames = varnames[:40] + ["rest"]
-        n = 40 + 1
-
-    # if <10 vars, make it like 10
-    if n < 10:
-        n_extra = 10 - n
-        imps_avg = np.append(imps_avg, [0.0] * n_extra)
-        imps_stddev = np.append(imps_stddev, [0.0] * n_extra)
-        varnames = varnames + [""] * n_extra
-        n = 10
-
-    # put in percent scales
-    imps_avg = imps_avg * 100.0
-    imps_stddev = imps_stddev * 100.0
-
-    # start fig
-    if fig_ax is None:
-        # so labels are above lines. Must be before figure()
-        plt.rcParams["axes.axisbelow"] = False
-
-        fig, ax = plt.subplots()
-    else:
-        fig, ax = fig_ax
-        ax.cla()  # clear axis
-
-    # plot
-    ytick_ylocs = np.arange(n)  # eg [0, 1, 2, .., 9] for 10 vars
-
-    ax.xaxis.grid(visible=True, color="0.9", linestyle="--", linewidth=1)
-    bar_lw = 0.2 if n < 15 else 0.3
-    err_lw = 3 if n < 15 else 1
-    ax.barh(
-        ytick_ylocs,
-        imps_avg,
-        color="0.5",
-        height=bar_lw,
-        xerr=imps_stddev * 2,
-        error_kw={"ecolor": "0.9", "lw": err_lw, "capsize": 0, "capthick": 0},
-        align="center",
+    var_imps, errors = d.model.importance_per_var(include_stddev=True)
+    labels = d.colnames
+
+    var_imps = [100 * a for a in var_imps]
+    errors = [100 * a for a in errors]
+
+    df = pd.DataFrame(var_imps, columns=["importance"])
+    df["label"] = labels
+    df["erorrs"] = errors
+    df["low"] = df["importance"] - df["erorrs"] * 2
+    df["high"] = df["importance"] + df["erorrs"] * 2
+    df.sort_values(by=["importance"], inplace=True, ascending=True)
+
+    chart = (
+        alt.Chart(df)
+        .mark_bar()
+        .encode(
+            x=alt.X("importance", title="Relative importance (%)"),
+            y=alt.Y("label", title=None, sort="-x", axis=alt.Axis(labelLimit=200)),
+        )
+        .properties(title="Variable importances")
```

Comment:
spent some extra time reviewing this code block, i think the logic checks out although i guess we're not truncating anymore based on importance

assuming this, i think it looks good

Reply:
not truncating anymore since there is no need for the graph to "fit" inside a small container. I can not guarantee it looks "perfect" for edge cases like e.g. an enormous amount of variables, but it won't overlap any text. Also, in the future we might be selecting for specific variables to plot, so this logic will probably be something else entirely.